### PR TITLE
Cleanup configuration from unused code and warnings

### PR DESCRIPTION
**Pull request by klaren**

### CODE REVIEW

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java [hunk 1]
```diff
@@ -561,14 +560,14 @@ private void warnAboutDeprecations( Map<String,String> userSettings )
...
-        return httpConnectors( params );
+        return httpConnectors( params ).collect( Collectors.toList() );
     }
 
     /**
      * @return list of all configured http connectors
      */
     @Nonnull
-    public static List<HttpConnector> httpConnectors( @Nonnull Map<String,String> params )
+    private static Stream<HttpConnector> httpConnectors( @Nonnull Map<String,String> params )
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java]: Javadoc could use an update reflecting this change.

File: community/dbms/src/main/java/org/neo4j/commandline/dbms/LoadCommandProvider.java [hunk 2]
```diff
@@ -35,12 +36,14 @@ public LoadCommandProvider()
     }
 
     @Override
+    @Nonnull
```
Comment (burqen) [inline community/dbms/src/main/java/org/neo4j/commandline/dbms/LoadCommandProvider.java:39]: Nice catch!

File: community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseFactoryState.java [hunk 3]
```diff
@@ -91,6 +92,11 @@ public void addKernelExtensions( Iterable<KernelExtensionFactory<?>> newKernelEx
         }
     }
 
+    /**
+     * @deprecated This method has no side effects now since we moved to service loading instead, {@link LoadableConfig}
+     * should be used.
+     */
+    @Deprecated
```
Comment (burqen) [inline community/kernel/src/main/java/org/neo4j/graphdb/factory/GraphDatabaseFactoryState.java:99]: Nice catch! 🎉

**Review by burqen — CHANGES_REQUESTED** [review]
Looks good! Awaiting fix for failing `docscompile` and javadco fix.

File: community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java (hunk 1 above)
Comment (klaren) [reply community/kernel/src/main/java/org/neo4j/kernel/configuration/Config.java]: Oops, missed that. Nice catch too!

**Comment by klaren** [comment]
@burqen New commit available for reviewing

**Review by burqen — APPROVED** [review]
PR approved if build goes green 🎉

**Pull request merged**